web/src/features/otel/server/index.ts

```diff
@@ -1,7 +1,57 @@
-import { type IngestionEventType } from "@langfuse/shared/src/server";
+import {
+  type TraceEventType,
+  type IngestionEventType,
+} from "@langfuse/shared/src/server";
 import { randomUUID } from "crypto";
 import { ForbiddenError, ObservationLevel } from "@langfuse/shared";
 import { LangfuseOtelSpanAttributes } from "./attributes";
+import { redis } from "@langfuse/shared/src/server";
+
+export async function getSeenTracesSet(
+  resourceSpans: unknown,
+  projectId: string,
+): Promise<Set<string>> {
+  if (!redis) {
+    console.warn("Redis client not available");
+
+    return new Set();
+  }
+
+  const traceIds: Set<string> = new Set();
+  if (Array.isArray(resourceSpans)) {
+    resourceSpans.forEach((resourceSpan) => {
+      for (const scopeSpan of resourceSpan?.scopeSpans ?? []) {
+        for (const span of scopeSpan?.spans ?? []) {
+          traceIds.add(
+            Buffer.from(span.traceId?.data ?? span.traceId).toString("hex"),
+          );
+        }
+      }
+    });
+  }
+
+  const results = await Promise.all(
+    [...traceIds].map(async (traceId) => {
+      const key = `project:${projectId}:trace:${traceId}:seen`;
```

Comment:
I'd also add a `langfuse:` prefix. It's not necessary but may make co-uses of the redis db cleaner.